erigon-lib/diagnostics/sys_info.go

```diff
@@ -20,16 +20,27 @@ var (
 
 func (d *DiagnosticClient) setupSysInfoDiagnostics() {
 	sysInfo := GetSysInfo(d.dataDirPath)
-	if err := d.db.Update(d.ctx, RAMInfoUpdater(sysInfo.RAM)); err != nil {
-		log.Error("[Diagnostics] Failed to update RAM info", "err", err)
-	}
+	err := d.db.Update(d.ctx, func(tx kv.RwTx) error {
+		err := RAMInfoUpdater(d.hardwareInfo.RAM)(tx)
+		if err != nil {
```

Comment:
This check makes no sense because returned error goes to scope of caller
Function from .Update which also may be not what you wanted. 

I would recommend just make 'return fn()' inside db.Update and check error outside

Also err has to be checked after each call